lib/src/package_meta.dart

```diff
@@ -22,7 +22,8 @@ abstract class PackageMeta {
 
   /// Use this instead of fromDir where possible.
   factory PackageMeta.fromElement(LibraryElement libraryElement) {
-    if (libraryElement.isInSdk) return new PackageMeta.fromDir(getSdkDir());
+    // Workaround for dart-lang/sdk#32707.  Replace with isInSdk once that works.
+    if (libraryElement.source.uri.scheme == 'dart') return new PackageMeta.fromDir(getSdkDir());
```

Comment:
You could also use `libraryElement.source.isInSystemLibrary` which is the same logic as this behind a named getter.

https://github.com/dart-lang/build/pull/986/files

That's what the build system and DDC use.